### DISCUSSION TITLE
ArrayField is not exported anymore after 6.x

### DISCUSSION THREAD

**Discussion by LasaleFamine** [react-hook-form/react-hook-form#2174] [General] [answered]
Before the 6.x version I was able to import the type `ArrayField` for typing my inner components that would access fields from `useArrayFields`. After the 6.x the type is not exported anymore or at least I can see it exported from `react-hook-form/dist/types/form` which doesn't seem the correct place.
Is this a design choice? How can I type fields like I did before? Something like:

```
fields: Partial<ArrayField<MyTypeForFields>>
```

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
Yes, it's by design, we are taking type more serious now and documented them down properly for those that we exported. 

https://react-hook-form.com/ts/

@kotarella1110 what your opinion on export this type.

**Reply by LasaleFamine** [reply]
Hi @bluebill1049 and thanks for the response. I think that type was actually helpful and somehow required for me, since I have made different components which have shared the `FormContext` and I need to pass down what I get from the `useFieldArray` hook. With the current 6.x I'm afraid I need to rebuild the type `ArrayField` for my good. 

Do you think I can find a better solution with the types you are exposing right now? I'm currently stuck with this problem coming from Typescript when the application just works correctly.

**Reply by bluebill1049** [reply]
you can direct hit the form.ts file for now and I am waiting for our ts master @kotarella1110 to confirm whether we should make it "public" and document it down.

**Reply by kotarella1110** [reply]
@LasaleFamine Can you provide usage example in codesandbox?

**Reply by LasaleFamine** [reply]
Hi @kotarella1110, of course: https://codesandbox.io/s/react-hook-form-arrayfields-v57vm 

I've tried to be liege to my project so you can understand better my problem: the `ActionSelect` component is the one where I'm using the `ArrayField` type.
Thanks for your support!

**Reply by kotarella1110** [reply]
@LasaleFamine @bluebill1049 let's export this type! I will do this tonight!

**Reply by bluebill1049** [reply]
PR ready https://github.com/react-hook-form/react-hook-form/pull/2254/files
